pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -1427,31 +1475,38 @@ func (gce *GCECloud) ExternalID(instance string) (string, error) {
 	}
 
 	// Fallback to GCE API call if metadata server fails to retrieve ID
-	inst, err := gce.getInstanceByName(instance)
+	inst, err := gce.resolveHost(instance)
 	if err != nil {
 		return "", err
 	}
-	return strconv.FormatUint(inst.Id, 10), nil
+	return strconv.FormatUint(inst.ID, 10), nil
 }
 
 // InstanceID returns the cloud provider ID of the specified instance.
 func (gce *GCECloud) InstanceID(instance string) (string, error) {
-	return gce.projectID + "/" + gce.zone + "/" + canonicalizeInstanceName(instance), nil
+	host, err := gce.resolveHost(instance)
```

Comment:
Please be consistent with the use of 'host' and 'instance' here. I prefer instance, although I understand the potential name clash problem that perhaps you were trying to solve.  How about:

```
instanceObj, err := gce.getInstanceByName(instance)
```
?